File: apps/mobile/src/types/analysis.ts
import type { BadgeTone } from "@/components/ui/Badge";

export type StatusTone = BadgeTone;

// Shared shape consumed by ScriptView/ScriptSentenceRow regardless of which
// metric (pace vs energy) the sentence array came from.
export type SyncedSentence = {
  text: string;
  startTime: number;
  endTime: number;
  status: string;
};

export type PaceSentence = SyncedSentence & { status: "fast" | "slow" | "stable" };
export type EnergySentence = SyncedSentence & { status: "high" | "low" | "stable" };

export type PaceMetric = {
  wpm: number;
  statusLabel: string;
  tone: StatusTone;
  description: string;
  sentences: PaceSentence[];
};

export type EnergyMetric = {
  score: number;
  title: string;
  tone: StatusTone;
  description: string;
  sentences: EnergySentence[];
};

export type FillerWord = { word: string; timestamp: string };

export type FillerWordsMetric = {
  count: number;
  tone: StatusTone;
  words: FillerWord[];
  tip: string;
};

export type ContentLogicMetric = {
  statusLabel: string;
  tone: StatusTone;
  strengthTitle: string;
  strengthDescription: string;
  tipTitle: string;
  tipDescription: string;
};

export type FeedbackResult = {
  globalScore: number;
  statusLabel: string;
  summary: string;
  pace: PaceMetric;
  energy: EnergyMetric;
  fillerWords: FillerWordsMetric;
  contentLogic: ContentLogicMetric;
};

File: apps/backend/app/schemas/analysis.py
from typing import Literal

from pydantic import BaseModel

Tone = Literal["success", "warning", "error"]


class PaceSentence(BaseModel):
    text: str
    startTime: float
    endTime: float
    status: Literal["fast", "slow", "stable"]


class PaceAnalysis(BaseModel):
    wpm: int
    statusLabel: str
    tone: Tone
    description: str
    sentences: list[PaceSentence]


class EnergySentence(BaseModel):
    text: str
    startTime: float
    endTime: float
    status: Literal["high", "low", "stable"]


class EnergyAnalysis(BaseModel):
    score: int
    title: str
    tone: Tone
    description: str
    sentences: list[EnergySentence]


class FillerWordItem(BaseModel):
    word: str
    timestamp: str


class FillerWordsAnalysis(BaseModel):
    count: int
    tone: Tone
    words: list[FillerWordItem]
    tip: str


class ContentLogicAnalysis(BaseModel):
    statusLabel: str
    tone: Tone
    strengthTitle: str
    strengthDescription: str
    tipTitle: str
    tipDescription: str


class SpeechAnalysisResult(BaseModel):
    globalScore: int
    statusLabel: str
    summary: str
    pace: PaceAnalysis
    energy: EnergyAnalysis
    fillerWords: FillerWordsAnalysis
    contentLogic: ContentLogicAnalysis
